src/main/java/org/folio/services/domainevent/ItemDomainEventService.java

```diff
@@ -0,0 +1,163 @@
+package org.folio.services.domainevent;
+
+import static io.vertx.core.Future.succeededFuture;
+import static org.apache.logging.log4j.LogManager.getLogger;
+import static org.folio.rest.support.ResponseUtil.isCreateSuccessResponse;
+import static org.folio.rest.support.ResponseUtil.isDeleteSuccessResponse;
+import static org.folio.rest.support.ResponseUtil.isUpdateSuccessResponse;
+import static org.folio.services.kafka.topic.KafkaTopic.INVENTORY_ITEM;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.core.Response;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.ImmutableTriple;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.commons.lang3.tuple.Triple;
+import org.apache.logging.log4j.Logger;
+import org.folio.persist.HoldingsRepository;
+import org.folio.persist.ItemRepository;
+import org.folio.rest.jaxrs.model.HoldingsRecord;
+import org.folio.rest.jaxrs.model.Item;
+import org.folio.services.batch.BatchOperationContext;
+import org.folio.services.item.effectivevalues.ItemWithHolding;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+
+public class ItemDomainEventService {
+  private static final Logger log = getLogger(ItemDomainEventService.class);
+
+  private final ItemRepository itemRepository;
+  private final HoldingsRepository holdingsRepository;
+  private final BaseDomainEventService<Item> domainEventService;
```

Comment:
The `Base` prefix usually suggests to me that it is intended to be used as a base class for an inheritance hierarchy.

This class appears to be a collaborator rather than a part of an inheritance hierarchy, could that collaboration be better described with a different name?

Reply:
I'll add `Common` prefix, if that's fine.